### ISSUE TITLE
Faster MREW for DEB ?

### ISSUE THREAD

**Issue by c-michail** [spinettaro/delphi-event-bus#62]
Hello,
   Take a look at the following articles! I think DEB can benefit from this. (it has 4 parts)
   https://www.thedelphigeek.com/2020/11/readers-writer-lock-part-1-why.html

Thank you

**Comment by wxinix** [comment]
The article is great but it doesn't say TLightweightMREW is a clear winner than TMREWSync which is used by DEB?

**Comment by dkounal** [comment]
https://www.thedelphigeek.com/2021/02/readers-writer-lock-part-3-some-numbers.html

**Comment by spinettaro** [comment]
@chmichael I agree with @wxinix  . From the article we cannot establish a clear winner as also reported in part 3. @chmichael do you have some numbers to highlight the fact the new TLightMREW is much better for this project?

**Comment by c-michail** [comment]
@spinettaro @wxinix Did you read all the 4 parts ? (i posted only the 1st part, the other is on main page)
As @dkounal Part 3 and Part 4 has some numbers!

Do you we have a benchmark for DEB ?

**Comment by wxinix** [comment]
I did read all parts. It is a great article by a respected Delphi guru. And thank you for referencing those articles.

Actually at one time DEB (for my private fork) did use the latest TLightweightMREW, which comes with Sydney 10.4 (in its release note it was noted for its slightly better performance)

But when I submitted my PR to the main DEB repo, I sticked to the old MREW, mainly for compatibility with older versions of Delphi, considering TLightweightMREW only works for the latest. 

We can do a conditional compile of course. But the number in the article shows mixed benchmarking results —— it doesn’t seem to suggest a clear winner.

**Comment by dkounal** [comment]
I think that only windows can get a better speed but it is safe only with TLightweightMREW**x** and only in a number of scenarios

**Comment by c-michail** [comment]
What ever you want guys but in my opinion it will be nice to have that option too! :)

**Comment by c-michail** [comment]
Take a look:
https://components4developers.blog/forums/topic/faster-mrew/
